### PR TITLE
Ports dbcore refactor, and limits the maximum amount of concurrent async queries to a variable amount

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/tgstation/pull/59676

## About The Pull Request

Refactors dbcore to work off a subsystem if executed async and limits the maximum amount of concurrent async queries to 25.

## Why It's Good For The Game

TG coders say: 

> May or may not resolve terry crashes, however, each query creates a new thread which takes up 2mb, preventing the game from using that 2mb. This can lead to ooms if they stack up, e.g. due to poor connectivity. This solves that issue.

Performance gain

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

Setup and game start (No SQL or DB errors)

image

Running empty queries with DB connected

image

</details>

## Changelog
:cl: rkz, Watermelon914
server: Refactors dbcore.dm and limits the maximum amount of concurrent async queries to 25.
/:cl:

### CODE REVIEW

**Review by Crossedfall — COMMENTED** [review]
We don't actually need this since we haven't had any of the problems they have. That being said, it _could_ provide a minimal bump to general query performance?

I'm a bit curious what prompted for the port

**Comment by Tsar-Salat** [comment]
> I'm a bit curious what prompted for the port

I'm doing a general push to make the lobby windows(changelog, manifest, latejoin) and interface (observe, etc) TGUI instead of html. Had discussions with maints for possible design directions.

I want to nip any of the pitfalls or performance issues TG had in the bud before I decide on where to go. This is one of them.

**Review by Crossedfall — APPROVED** [review]
I don't like the end goal. But just looking at the PR as a thing, it seems fine as far as I can tell.

**Review by itsmeow — APPROVED** [review]
Code looks fine to me.

File: code/controllers/subsystem/time_track.dm [hunk 1]
```diff
@@ -140,13 +143,19 @@ SUBSYSTEM_DEF(time_track)
 			length(SSair.hotspots),
 			length(SSair.networks),
 			length(SSair.high_pressure_delta),
+			//length(SSair.active_super_conductivity), // LINDA, I assume
```
Comment (itsmeow) [inline code/controllers/subsystem/time_track.dm:145]: we call this thermal transfer/thermal conductivity, not superconductivity. Might still exist

File: code/modules/admin/verbs/debug.dm [hunk 2]
```diff
@@ -790,6 +790,24 @@ But you can call procs that are of type /mob/living/carbon/human/proc/ for that
...
+		var/datum/db_query/query = SSdbcore.NewQuery("NULL")
+		INVOKE_ASYNC(query, TYPE_PROC_REF(/datum/db_query, Execute))
+		queries += query
+
+	for(var/datum/db_query/query as anything in queries)
+		query.sync()
+		qdel(query)
+	queries.Cut()
+
+	message_admins("[key_name_admin(src)] ran [val] empty queries.")
```
Comment (itsmeow) [inline code/modules/admin/verbs/debug.dm:809]: Why?

Comment (Tsar-Salat) [reply code/modules/admin/verbs/debug.dm:809]: why what

File: code/controllers/subsystem/time_track.dm (hunk 1 above)
Comment (Tsar-Salat) [reply code/controllers/subsystem/time_track.dm:145]: Is it important that we track it? 

I will add it back if so, I just assumed we didnt really care about it

**Comment by Rukofamicom** [comment]
I'm afraid to touch this one, it's beyond my field of knowledge. 
Get with Crossed/Kat to merge when they're available.

**Review by itsmeow — APPROVED** [review]

**Comment by Tsar-Salat** [comment]
Ive checked with the new banning panel, dont see any kinks.

image

**Pull request merged**